File: Code/0258/0258.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
文件名: PersonalLeetCode
作者: Bolun Xu
创建日期: 2025
版本: 1.0
描述: 各位相加 -- 硬循环。
时间复杂度： O(Logn)
空间复杂度： O(Logn)

"""


class Solution:
    def addDigits(self, num: int) -> int:
        def dfs(n: int) -> int:
            s = 0
            for i in str(n):
                s += int(i)
            n = s

            if n < 10:
                # print(num)
                return n
            else:
                return dfs(n)

        ans = dfs(num)
        return ans


if __name__ == '__main__':
    sol = Solution()
    print(sol.addDigits(38))

File: Code/1000/0258/0258.cpp
/******************************************************************************
 * 文件名: Code
 * 作者: BOLUN XU
 * 创建日期: 2025
 * 版本: 1.0
 * 描述: 各位相加。
 * 时间复杂度： O(1)
 * 空间复杂度： O(1)
 ******************************************************************************/
#include <iostream>

using namespace std;

class Solution {
public:
    int addDigits(int num) {
        return (num - 1) % 9 + 1;
    }
};

int main() {
  Solution s;
  cout << s.addDigits(38) << endl;
}